src/procstat.c

```diff
@@ -736,12 +737,29 @@ void procstat_u64_series_add_point(struct procstat_series_u64 *series, uint64_t
 	int64_t delta;
 	int64_t delta2;
 	int64_t avg_delta;
-	unsigned reset;
+	unsigned reset, reset_interval_expired;
+	uint64_t reset_interval;
+
+	reset_interval = __atomic_load_n(&series->reset_interval, __ATOMIC_RELAXED);
+
+	struct timespec cur_time;
+	if (clock_gettime( CLOCK_REALTIME, &cur_time) == -1) {
+		return -1;
+	}
+
+	uint64_t time_since_last_reset = cur_time.tv_sec - series->last_reset_time;
+
+	if ((reset_interval > 0)&&(time_since_last_reset > reset_interval)) {
+		reset_interval_expired = 1;
```

Comment:
lets use `bool` for this variable.

Reply:
OK, but I used the same pattern as 'reset'.